pkg/monitoring/domainstats/downwardmetrics/scraper.go

```diff
@@ -55,6 +51,13 @@ func (s *Scraper) Scrape(socketFile string, vmi *k6sv1.VirtualMachineInstance) {
 	}
 	defer cli.Close()
 
+	version, err := cli.GetQemuVersion()
```

Comment:
I think this has update implications. We probably have to investigate the returned error to find out if the remote virt-launcher server returned an "unimplemented" error. Otherwise we would break existing metrics collection until the VM is restarted or migrated.

Reply:
Can this be tested by: from the current build downgrading kubevirt to a previous version, creating a VM and checking the returned error? If not, how can I test it?

Reply:
yes that would be the way to go (you can just reference in the KubeVirt CR an older release when you have the newer operator installed).

Reply:
I used this test to check this behavior: https://github.com/kubevirt/kubevirt/commit/65559dc0cc8e0eb9d4e79684a25b076f1996f34a. I'm getting the following error: `{"component":"virt-handler","level":"error","msg":"failed to update qemu stats from socket //pods/2fb30bea-f00f-43e6-b298-6d8e33d2d59d/volumes/kubernetes.io~empty-dir/sockets/launcher-sock","pos":"scraper.go:57","reason":"unknown error encountered sending command GetQemuVersion: rpc error: code = Unimplemented desc = unknown method GetQemuVersion for service kubevirt.cmd.v1.Cmd","timestamp":"2023-03-22T11:23:18.781782Z"}` Should we check if `Unimplemented` error is happening and ignore it or just throw a warning?

Reply:
Yes my suggestion would be this: first I think you should get a grpc error back which you can exactly identify (so not just string matching). Second just set a string like `qemu-unknown` or something as fallback in this case and log an info string.